DCOS/regression-testing/src/script/disable_linux_agents_dcos_metrics.sh

```diff
@@ -19,25 +19,27 @@ done
 IPS="$PRIVATE_IPS"
 
 if [[ -z $IPS ]]; then
-  return 0
+  exit 0
 fi
 
-declare -a cmds=(
-  "sudo systemctl stop dcos-metrics-agent.service || echo skipped"
-  "sudo systemctl stop dcos-metrics-agent.socket || echo skipped"
-  "sudo systemctl disable dcos-metrics-agent.service || echo skipped"
-  "sudo systemctl disable dcos-metrics-agent.socket || echo skipped"
-  "sudo systemctl stop dcos-checks-poststart.timer || echo skipped"
-  "sudo systemctl stop dcos-checks-poststart.service || echo skipped"
-  "sudo systemctl disable dcos-checks-poststart.timer || echo skipped"
-  "sudo systemctl disable dcos-checks-poststart.service || echo skipped")
+master_scp="scp -i ${SSH_KEY} -o ConnectTimeout=30 -o StrictHostKeyChecking=no -P 2200"
+master_ssh="ssh -i ${SSH_KEY} -o ConnectTimeout=30 -o StrictHostKeyChecking=no -p 2200 azureuser@${INSTANCE_NAME}.${LOCATION}.cloudapp.azure.com"
+agent_ssh="ssh -i /tmp/id_rsa -o ConnectTimeout=30 -o StrictHostKeyChecking=no"
 
-scp -i ${SSH_KEY} -o ConnectTimeout=30 -o StrictHostKeyChecking=no -P 2200 ${SSH_KEY} azureuser@${INSTANCE_NAME}.${LOCATION}.cloudapp.azure.com:/tmp/id_rsa
+$master_scp ${SSH_KEY} azureuser@${INSTANCE_NAME}.${LOCATION}.cloudapp.azure.com:/tmp/id_rsa
 
-ssh_cmd="ssh -o ConnectTimeout=30 -o StrictHostKeyChecking=no"
 for IP in $IPS; do
-  for CMD in "${cmds[@]}"; do
-    $ssh_cmd -i ${SSH_KEY} -p 2200 azureuser@${INSTANCE_NAME}.${LOCATION}.cloudapp.azure.com "$ssh_cmd -i /tmp/id_rsa azureuser@$IP $CMD"
-  done
+  $master_ssh "$agent_ssh azureuser@$IP sudo systemctl stop dcos-metrics-agent.socket"
+  $master_ssh "$agent_ssh azureuser@$IP sudo systemctl disable dcos-metrics-agent.socket"
+  $master_ssh "$agent_ssh azureuser@$IP sudo systemctl stop dcos-metrics-agent.service"
+  $master_ssh "$agent_ssh azureuser@$IP sudo systemctl disable dcos-metrics-agent.service"
+  json=$($master_ssh "$agent_ssh azureuser@$IP cat /opt/mesosphere/etc/dcos-diagnostics-runner-config.json")
```

Comment:
Could you also stop the diagnostics service while changing the config file, before restart it?